src/warp.jl

```diff
@@ -74,10 +74,13 @@ julia> indices(imgr)
 ```
 """
 function warp_new{T}(img::AbstractExtrapolation{T}, tform, inds::Tuple = autorange(img, inv(tform)))
-    out = OffsetArray(Array{T}(map(length, inds)), inds)
+    out = _allocate_array(T, inds)
     warp!(out, img, tform)
 end
 
+_allocate_array{T,N}(::Type{T}, inds::NTuple{N,Base.OneTo}) = Array{T}(map(length, inds))
+_allocate_array{T,N}(::Type{T}, inds::NTuple{N,AbstractUnitRange}) = OffsetArray(Array{T}(map(length, inds)), inds)
```

Comment:
unsure if this change could be considered controversial, but I did it because now behaviour should is consistent between `warp(...)` and `collect(warpedview(...))`. It does negatively affect type inference

```julia
julia> summary(warp(img_camera, tfm, indices(img_camera)))
"512×512 Array{Gray{N0f8},2}"

julia> summary(collect(warpedview(img_camera, tfm, indices(img_camera))))
"512×512 Array{Gray{N0f8},2}"
```

and 

```julia
julia> summary(warp(img_camera, tfm))
"-78:591×-78:591 OffsetArray{Gray{N0f8},2}"

julia> summary(collect(warpedview(img_camera, tfm)))
"-78:591×-78:591 OffsetArray{Gray{N0f8},2}"
```